docs/components/gantry/oneaxis.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "Configure a One-Axis Gantry"
+linkTitle: "oneaxis"
+weight: 70
+type: "docs"
+description: "Configure a oneaxis gantry."
+tags: ["gantry", "components"]
+# SME: Rand
+---
+
+Configure a `oneaxis` gantry to integrate a one-axis gantry into your robot:
+
+{{< tabs >}}
+{{% tab name="Config Builder" %}}
+
+Navigate to the **config** tab of your robot's page in [the Viam app](https://app.viam.com).
+Click on the **Components** subtab and navigate to the **Create component** menu.
+Enter a name for your sensor, select the type `gantry`, and select the `oneaxis` model.
+
+Click **Create component**.
+Paste into the **Attributes** box:
+
+``` json
+"motor": <"your-motor-name">,
+"length_mm": <int>,
+"axis": {
+    "X": <int>,
+    "Y": <int>,
+    "Z": <int>
+}
+```
+
+![Creation of a one-axis gantry component in the Viam app config builder.](../img/oneaxis-ui-config.png)
+
+Edit and fill in the attributes as applicable.
+
+{{% /tab %}}
+{{% tab name="JSON Template" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  "components": [
+    // your motor config
+    {
+      "name": <"your-oneaxis-gantry-name">,
+      "type": "gantry",
+      "model": "oneaxis",
+      "attributes": {
+        "motor": <"your-motor-name">,
+        "length_mm": <int>,
+        "axis": {
+          "X": <int>,
+          "Y": <int>,
+          "Z": <int>
+        }
+      },
+      "depends_on": []
+    }
+  ]
+}
+```
+
+{{% /tab %}}
+{{% /tabs %}}
+
+The following attributes are available for `oneaxis` gantries:
+
+| Attribute | Inclusion | Description |
+| ----------- | -------------- | --------------  |
+| `length_mm` | **Required** | Length of the axis of the gantry in millimeters. |
+| `motor` | **Required** | `name` of the [motor](/components/motor) that moves the gantry's actuator. |
+| `axis` | **Required** | The translational axis for the gantry. Must be exactly one of x, y, or z. |
+| `board`  |  Optional | `name` of the [board](/components/board) containing the limit switches and pins. If `limit_pins` exist, `board` is required. |
+| `limit_pins`  | Optional | The `boards`'s pins attached to the limit switches on either end. If motor type is [stepper](/components/motor/gpiostepper/), `limit_pins` are required. |
```

Comment:
```suggestion
| `limit_pins`  | Optional | The `boards`'s pins attached to the limit switches on either end. If motor type is a [stepper](/components/motor/gpiostepper/), `limit_pins` are required. |
```
Is this true if it's a TMC or DMC model that includes stallguard/bump detection/whatever it's called? If not, change [stepper] to [gpiostepper]. If so, need to link to all 3 :o

Reply:
I do not know, was adapting from former version that specifies GPIOstepper. Should we call in Rand? She is OOO today but I would be down to page her for review here.

Reply:
@JessamyT Talked to steve and he said that he believes it is encoder-dependent (see update), but that we should check in with rand again to be sure about accuracy one she's back (won't be till the 2nd, so I'm in favor of landing this for now)